### PR TITLE
Add support for listening to messages as they are appended #954

### PR DESCRIPTION
**Pull request by peter-lawrey**

### CODE REVIEW

File: src/main/java/net/openhft/chronicle/queue/impl/single/SingleChronicleQueue.java [hunk 1]
```diff
@@ -164,13 +165,14 @@ protected SingleChronicleQueue(@NotNull final SingleChronicleQueueBuilder builde
             storeFactory = builder.storeFactory();
             checkInterrupts = builder.checkInterrupts();
             metaStore = builder.metaStore();
-            doubleBuffer = false; //builder.doubleBuffer();
+            doubleBuffer = builder.doubleBuffer();
```
Comment (nicktindall) [inline src/main/java/net/openhft/chronicle/queue/impl/single/SingleChronicleQueue.java:172]: Didn't this have a race condition in it?

File: src/test/java/net/openhft/chronicle/queue/impl/single/SingleChronicleQueueTest.java [hunk 2]
```diff
@@ -3244,6 +3277,71 @@ protected SingleChronicleQueueBuilder builder(@NotNull File file, @NotNull WireT
...
+                    appenderListenerDump.append("idx: ").append(Long.toHexString(idx)).append("\n");
+                    w.bytes().readSkip(-4);
+                    appenderListenerDump.append(Wires.fromSizePrefixedBlobs(w)).append("\n");
+                })
+                .testBlockSize();
+    }
+
     @NotNull
     protected SingleChronicleQueueBuilder binary(@NotNull File file) {
         return builder(file, WireType.BINARY_LIGHT);
```
Comment (nicktindall) [inline src/test/java/net/openhft/chronicle/queue/impl/single/SingleChronicleQueueTest.java:3356]: This diff is impossible to review... is there a way we could do it with less change?

File: src/main/java/net/openhft/chronicle/queue/impl/single/StoreAppender.java [hunk 3]
```diff
@@ -630,7 +630,9 @@ private void writeBytesInternal(@NotNull final BytesStore bytes, boolean metadat
             openContext(metadata, safeLength);
 
             try {
-                writeContext.wire().bytes().write(bytes);
+                final Bytes<?> bytes0 = writeContext.wire().bytes();
+                bytes0.readPosition(bytes0.writePosition());
```
Comment (nicktindall) [inline src/main/java/net/openhft/chronicle/queue/impl/single/StoreAppender.java:639]: Why do we set the `readPosition` here?

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -34,4 +34,4 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}  # Needed to get PR information, if any
           SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
-        run: mvn -B verify org.sonarsource.scanner.maven:sonar-maven-plugin:sonar -Dsonar.projectKey=OpenHFT_Chronicle-Queue
+        run: mvn -B verify org.sonarsource.scanner.maven:sonar-maven-plugin:sonar -Dsonar.projectKey=OpenHFT_Chronicle-Queue -Psonar -Pjava11
```
Comment (JerryShea) [inline .github/workflows/build.yml:37]: why -Pjava11?

**Pull request closed without merging**